common/utils.c

```diff
@@ -428,13 +428,24 @@ TDNFYesOrNo(
 
     if(!pArgs->nAssumeYes && !pArgs->nAssumeNo)
     {
-        pr_crit("%s", pszQuestion);
-        while ((opt = getchar()) == '\n' || opt == '\r');
-        opt = tolower(opt);
-        if (opt != 'y' && opt != 'n')
-        {
-            dwError = ERROR_TDNF_INVALID_INPUT;
-            BAIL_ON_TDNF_ERROR(dwError);
+        while(1) {
+            pr_crit("%s", pszQuestion);
+            char buf[256];
+            char *ret;
+
+            ret = fgets(buf, sizeof(buf)-1, stdin);
+            if (ret != buf || strlen(buf) == 0) {
+                /* should not happen */
+                dwError = ERROR_TDNF_INVALID_INPUT;
+                BAIL_ON_TDNF_ERROR(dwError);
+            }
+            buf[strlen(buf)-1] = 0;
+            if (strcasecmp(buf, "yes") == 0 || strcasecmp(buf, "y") == 0 ||
```

Comment:
nit:
```
char strs[][4] = {"yes", "y", "no", "n"};
for (i = 0; i < 4; ++i) {
  if (strcasecmp(buf, strs[i]) break;
}

if (i >= 4) {
  handle answers other than strs
}
```

Reply:
Nice idea if we had more possible answers, but with just 4 we don't save any lines.